lib/vty.c

```diff
@@ -345,8 +345,17 @@ int vty_out(struct vty *vty, const char *format, ...)
 	case VTY_SHELL_SERV:
 	case VTY_FILE:
 	default:
+		vty->vty_buf_size_written += strlen(filtered);
 		/* print without crlf replacement */
 		buffer_put(vty->obuf, (uint8_t *)filtered, strlen(filtered));
+		/* For every chunk of memory, we invoke vtysh_flush where we
+		 * put the data of collective vty->obuf Linked List items on the
+		 * socket and free the vty->obuf data.
+		 */
+		if (vty->vty_buf_size_written >= MAX_INTERMEDIATE_FLUSH) {
+			vty->vty_buf_size_written = 0;
```

Comment:
still unclear about the meaning of this new variable.
it's set to zero here, even if nothing was written (or less than the amount of data pending), so it no longer represents the amount of data buffered?
it's set to zero (again) on a complete send.
what is it supposed to represent - it's not the "buf_size_written", right?

Reply:
The idea is, that we keep accumulating data (few vty_outs() calls) until we hit 32KB and once we hit the 32KB limit, 

- we call vtysh_flush where the data(32KB) is written into the socket
- reset vty->vty_buf_size_written to 0 so we again accumulate another 32KB.

vty_buf_size_written  is essentially to tell me how much memory is accumulated.  if renaming it makes life easy, would vty_buf_size_accumulated work?